### PR TITLE
feat(store-ui): Overlay Atom

### PR DESCRIPTION
**Pull request by igorbrasileiro**
## What's the purpose of this pull request?
Adds Overlay atom: component, test, and story.

### CODE REVIEW

File: packages/store-ui/src/atoms/Overlay/stories/Overlay.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      backgroundColor: 'rgba(0, 255, 0, 0.5)',
+    },
+    black: {
+      backgroundColor: 'rgba(0, 0, 0, 0.5)',
+    },
+  }
+
+  return (
+    <div style={{ height: 'calc(100vh + 150px)' }}>
+      A content outside me
```
Comment (tlgimenes) [inline packages/store-ui/src/atoms/Overlay/stories/Overlay.stories.tsx:27]: Maybe?
```suggestion
      Content outside of me
```

File: packages/store-ui/src/atoms/Overlay/stories/Overlay.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      <Component
+        {...props}
+        style={{
```
Comment (tlgimenes) [inline packages/store-ui/src/atoms/Overlay/stories/Overlay.stories.tsx:30]: Nice! I think we can wait for #828 so we can move these styles there, what do you think?

**Comment by igorbrasileiro** [comment]
Add role presentation https://www.w3.org/WAI/PF/HTML/wiki/RoleAttribute

File: packages/store-ui/src/atoms/Overlay/stories/Overlay.stories.tsx (hunk 1 above)
Comment (igorbrasileiro) [reply packages/store-ui/src/atoms/Overlay/stories/Overlay.stories.tsx:27]: 👀 I think `Content outside me` is better, no? https://dictionary.cambridge.org/pt/dicionario/ingles/outside?q=outside+of
Outside vs outside of in this case has no difference. Ain't an English expert, remember this. hahha

**Review by icazevedo — APPROVED** [review]

**Review by tlgimenes — APPROVED** [review]
LGTM, however, I feel the styling and storybook of this component requires a modal component

**Pull request merged**